### PR TITLE
Spectests

### PR DESCRIPTION
**Pull request by nalepae**
**What type of PR is this?**
Bug fix

**What does this PR do? Why is it needed?**
Implement spec tests for `get_custody_columns` and fix the implementation.

### CODE REVIEW

File: beacon-chain/core/peerdas/helpers.go [hunk 1]
```diff
@@ -23,10 +23,13 @@ const (
 
 	// Number of cells in the extended matrix
 	extendedMatrixSize = fieldparams.MaxBlobsPerBlock * cKzg4844.CellsPerExtBlob
+
+	// Maxmimum value of an uint64
+	maxUint64 = ^uint64(0)
```
Comment (nisdas) [inline beacon-chain/core/peerdas/helpers.go:28]: we should use the value from the std lib (math.MaxUint64)

File: beacon-chain/core/peerdas/helpers.go [hunk 2]
```diff
@@ -75,28 +83,43 @@ func CustodyColumnSubnets(nodeId enode.ID, custodySubnetCount uint64) (map[uint6
...
-		nextNodeIdUInt256.Add(nodeIdUInt256, uint256.NewInt(i))
-		nextNodeIdUInt64 := nextNodeIdUInt256.Uint64()
-		nextNodeId := bytesutil.Uint64ToBytesLittleEndian(nextNodeIdUInt64)
+	// Convert the node ID to a big int.
+	nodeIdUInt256 := new(uint256.Int).SetBytes(nodeId.Bytes())
 
-		hashedNextNodeId := hash.Hash(nextNodeId)
-		subnetId := binary.LittleEndian.Uint64(hashedNextNodeId[:8]) % dataColumnSidecarSubnetCount
+	// Handle the maximum value of a uint256 case.
+	if nodeIdUInt256.Cmp(maxUint256) == 0 {
```
Comment (nisdas) [inline beacon-chain/core/peerdas/helpers.go:90]: Why set it to 0 ? Do spec tests expect it

Comment (nisdas) [reply beacon-chain/core/peerdas/helpers.go:90]: Just realized they changed the implementation, 
https://github.com/ethereum/consensus-specs/blob/dev/specs/_features/eip7594/das-core.md#get_custody_columns

Comment (nalepae) [reply beacon-chain/core/peerdas/helpers.go:90]: I proposed a new change: https://github.com/ethereum/consensus-specs/pull/3742

File: beacon-chain/core/peerdas/helpers.go (hunk 1 above)
Comment (nalepae) [reply beacon-chain/core/peerdas/helpers.go]: Fixed in dc02b04dfe5524be4242ed7f47f3f653f06d8c40.

**Review by nisdas — APPROVED** [review]

**Review by nisdas — APPROVED** [review]

**Pull request merged**